### PR TITLE
[BEAM-2596] Split Java and Python precommit jobs

### PR DESCRIPTION
**Pull request by kennknowles**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

This is to mitigate the issue of timeouts currently happening, until we can completely move to the pipeline build.

### CODE REVIEW

**Comment by kennknowles** [comment]
R: anyone, since the build remains perma-red/gray
Then also R: @jasonkuster to devise the right next steps

File: .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy [hunk 1]
```diff
@@ -40,6 +40,19 @@ mavenJob('beam_PreCommit_Java_MavenInstall') {
   // Sets that this is a PreCommit job.
   common_job_properties.setPreCommit(delegate, 'Maven clean install')
 
-  // Maven goals for this job.
-  goals('-B -e -Prelease,include-runners,jenkins-precommit,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner -DrepoToken=$COVERALLS_REPO_TOKEN -DpullRequest=$ghprbPullId help:effective-settings clean install coveralls:report')
+  // Maven goals for this job: The Java SDK, its dependencies, and things that depend on it.
+  goals('''\
+    --batch-mode \
+    --errors \
+    --activate-profiles release,jenkins-precommit,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner \
```
Comment (tgroh) [inline .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy]: :100: for long-form flags

File: .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy [hunk 2]
```diff
@@ -40,6 +40,19 @@ mavenJob('beam_PreCommit_Java_MavenInstall') {
...
+    --projects sdks/java/core \
```
Comment (tgroh) [inline .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy]: I believe there exist modules that will be written in java and part of our repository that will not depend on the core SDK; I'm not sure how to meaningfully catch them, though

Comment (kennknowles) [reply .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy]: I don't have a great plan. I expect things to evolve from this crude step.

**Comment by kennknowles** [comment]
I think we should run the seed job from this PR first to make sure it succeeds, so that we can always run it from master if it does not.

**Comment by tgroh** [comment]
LGTM

File: .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy (hunk 2 above)
Comment (tgroh) [reply .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy]: Yeah, this is more important for the build pipeline stuff

**Comment by kennknowles** [comment]
Run Seed Job

**Comment by kennknowles** [comment]
retest this please

File: .test-infra/jenkins/job_beam_PreCommit_Python_MavenInstall.groovy [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  common_job_properties.setTopLevelMainJobProperties(
+    delegate,
+    'master',
+    150)
+
+  // Set Maven parameters.
+  common_job_properties.setMavenConfig(delegate)
+
+  // Sets that this is a PreCommit job.
+  common_job_properties.setPreCommit(delegate, 'Maven clean install')
```
Comment (kennknowles) [inline .test-infra/jenkins/job_beam_PreCommit_Python_MavenInstall.groovy]: Need to tweak the message here

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
retest this please

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
retest this please

**Comment by kennknowles** [comment]
OK, I can confirm that the tests are launched as expected. Now either of them may end up red but that is a different breakage. Since the seed job has already adjusted Jenkins, I will merge this so the status in the repo reflects what is running.

**Pull request merged**